### ISSUE TITLE
Unable to recurisly delete folder

### ISSUE THREAD

**Issue by Jimmy062006** [DiscUtils/DiscUtils#270]
The below code seems to generate a folder not empty Exception but I'm unsure why.

Would you have any idea's for me to try.

```csharp
try
{
    var info = vhdbNtfs.GetDirectoryInfo("AppData\\Local");
    if (info != null)
    {
        var dirs = info.GetDirectories();
        foreach (var dir in dirs)
        {
            dir.Delete(true);
        }
    }
}
catch(Exception ex)
{
    var error = ex;
}
```

Unable to delete non-empty directory

My understanding is that `Delete(true)` should recursively delete all folders/files.

**Comment by LTRData** [comment]
Yes that is how it is supposed to work. There seems to be something strange happening when you try to delete that directory and I can replicate the same issue here as well for that directory. I'll take a look and see what I can find.

**Comment by Jimmy062006** [comment]
I may have tracked it down to no permissions to delete certain things in the folder.

However this is a new lib to me so I just want to check that my thought train might be correct for it.

I guess its forced to obey the NTFS permission flags ? My guess is it's really getting an access denied but that is not caught ?

**Comment by LTRData** [comment]
No, it has nothing to do with permissions. Permissions are entirely ignored by this library. It rather seems to be related to some index cache bugs somewhere.

**Comment by LTRData** [comment]
No, sorry. You were right. I see now that `DiscDirectoryInfo.Delete(recursive: true)` fails to find hidden files and delete them, which causes this error. This behavior can be changed by setting `HideHiddenFiles = false` and `HideSystemFiles = false` on the `NtfsFileSystem.NtfsOptions` object. However, I still consider this a bug because those settings are ment to affect things like file/directory enumerations, a recursive directory delete operation should ignore these settings.

**Comment by Jimmy062006** [comment]
OK, so until that is patched I could just set `HideHiddenFiles = false` and `HideSystemFiles = false` for the folders that I am about to delete?

**Comment by LTRData** [comment]
Yes. That seems to work.

**Comment by Jimmy062006** [comment]
I can confirm that resolves my issue but will leave this open for you to deal with :)
